app/scripts/controllers/transactions.js

```diff
@@ -482,11 +527,43 @@ module.exports = class TransactionController extends EventEmitter {
         }
         if (txParams.blockNumber) {
           this.setTxStatusConfirmed(txId)
+          this._updateNonceDuplicates()
         }
       })
     })
   }
 
+  _updateNonceDuplicates () {
```

Comment:
we agreed that it would be easier to not maintain this list but just create in on demand
so we should create a new fn for nullifying txs from the same address with the same nonce